mkosi/installer/apt.py

```diff
@@ -39,11 +48,21 @@ def setup_apt(context: Context, repos: Sequence[str]) -> None:
             )
         )
 
-    sources = context.pkgmngr / "etc/apt/sources.list"
+    sources = context.pkgmngr / "etc/apt/sources.list.d/mkosi.sources"
     if not sources.exists():
         with sources.open("w") as f:
             for repo in repos:
-                f.write(f"{repo}\n")
+                f.write(
+                    textwrap.dedent(
+                        f"""\
+                        Types: {" ".join(repo.types)}
+                        URIs: {repo.url}
+                        Suites: {repo.suite}
+                        Components: {" ".join(repo.components)}
+                        {"Signed-By" if repo.signedby else "Trusted"}: {repo.signedby or "yes"}
+                        """
```

Comment:
The whole thing would be great as the `__str__` method on `AptRepository`—would save the `textwrap.dedent`, too.